### PR TITLE
[16.0][ADD] account_dms_field: New module

### PR DESCRIPTION
**Pull request by agent-z28**
This module adds "account_dms_field"

### CODE REVIEW

File: account_dms_field/models/dms_directory.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                childs = record.sudo().parent_id.child_directory_ids.name_get()
+            if list(
+                filter(
+                    lambda child: child[1] == record.name and child[0] != record.id,
+                    childs,
+                )
+            ):
+                raise ValidationError(
+                    _("A directory with the same name already exists.")
+                )
```
Comment (victoralmau) [inline account_dms_field/models/dms_directory.py]: This is not correct. If any changes need to be done, they should be done in `_prepare_directory_vals()` of `dms.field.template` and change the directory name there (replacing as necessary).

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: account_dms_field/models/dms_field_template.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    _inherit = "dms.field.template"
+
+    def _create_dms_directory(self):
+        """New method to define the directory automatically in demo data avoiding
+        error if it already exists."""
+        self.ensure_one()
+        if not self.dms_directory_ids:
+            self.with_context(
+                res_model=self._name, res_id=self.id
+            ).create_dms_directory()
```
Comment (victoralmau) [inline account_dms_field/models/dms_field_template.py]:
```suggestion
```

File: account_dms_field/demo/account_dms_data.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        <field name="name">Account</field>
+        <field name="storage_id" ref="dms.storage_demo" />
+        <field name="model_id" ref="account.model_account_move" />
+        <field name="user_field_id" ref="account.field_account_move__user_id" />
+        <field name="group_ids" eval="[(4, ref('read_access_account_move_group'))]" />
+    </record>
+
+    <function model="dms.field.template" name="_create_dms_directory">
+        <value eval="[ref('field_template_account')]" />
+    </function>
```
Comment (victoralmau) [inline account_dms_field/demo/account_dms_data.xml]:
```suggestion
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Since https://github.com/OCA/dms/pull/368 these changes are no longer necessary

File: account_dms_field/models/dms_directory.py (hunk 1 above)
Comment (agent-z28) [reply account_dms_field/models/dms_directory.py]: @victoralmau Thank you for the review, back at the desk i can fix it

Comment (agent-z28) [reply account_dms_field/models/dms_directory.py:30]: @victoralmau please check again

**Comment by agent-z28** [comment]
@bizzappdev please review

**Review by victoralmau — APPROVED** [review]

**Review by bizzappdev — APPROVED** [review]
Code and functional test LGTM

**Pull request merged**